doc/api/http.md

```diff
@@ -678,7 +678,7 @@ added: v0.3.0
 Reference to the underlying socket. Usually users will not want to access
 this property. In particular, the socket will not emit `'readable'` events
 because of how the protocol parser attaches to the socket. After
-`response.end()`, the property is nulled. The `socket` may also be accessed
+`request.end()`, the property is nulled. The `socket` may also be accessed
```

Comment:
Both seem incorrect. It's `null` when `ClientRequest` is instantiated via `http.request()` because a socket has not been assigned yet but it seems the property is never nulled after the `'socket'` event.

Reply:
I agree with @lpinca. It also seems something we might want to do (nulling the socket) not after `request.end()` but rather after the response has been fully consumed.

Reply:
hmmm... actually good point. Missed that on my initial read through